### PR TITLE
drop python 3.7 support and break importlib-metadata cycle

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by chrisburr** [comment]
@conda-forge/setuptools_scm Ready for review

**Comment by h-vetinari** [comment]
Thanks for fixing this! JFYI, the python 3.11 migration now depends on this PR through numpy -> importlib_metadata -> this fix.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -19,11 +19,10 @@ outputs:
       noarch: python
     requirements:
       host:
-        - python >=3.6
+        - python >=3.8
         - pip
       run:
-        - python >=3.6
-        - importlib-metadata
```
Comment (ocefpaf) [inline recipe/meta.yaml:26]: We should probably announce to pkg maintainers that it is fine to remove importlib_metadata now.

Comment (chrisburr) [reply recipe/meta.yaml:26]: It's not always, I maintain some packages which use Python 3.10+ features of `importlib.metadata` so you have to continue to use `importlib_metadata`.

Comment (ocefpaf) [reply recipe/meta.yaml:26]: Good to know. Thanks @chrisburr!

Comment (h-vetinari) [reply recipe/meta.yaml:26]: Another case in point: vanilla python 3.11 cannot run the numpy test suite without newer(-than-bundled) `importlib_metadata`, see [here](https://github.com/conda-forge/numpy-feedstock/pull/283#issuecomment-1291711115)

**Pull request merged**